apm-agent-core/src/test/java/co/elastic/apm/agent/MockTracer.java

```diff
@@ -102,4 +146,18 @@ public static ElasticApmTracer create(ConfigurationRegistry configurationRegistr
         when(tracer.getConfig(any())).thenAnswer(invocation -> configurationRegistry.getConfig(invocation.getArgument(0)));
         return tracer;
     }
+
+    public static class MockInstrumentationSetup {
+        public final ElasticApmTracer tracer;
+        public final MockReporter reporter;
+        public final ConfigurationRegistry config;
+        public final TestObjectPoolFactory objectPoolFactory;
```

Comment:
[minor] maybe provide getters for consistency, we don't have much direct field access in our codebase, and this part will be reused extensively into plugin tests.